### ISSUE TITLE
java.lang.IncompatibleClassChangeError: Structural change of android.support.v4.app.Fragment is hazardous ...

### ISSUE THREAD

**Issue by chenzhuo914** [googlecodelabs/android-topeka#8]
This erro is reproducible by following the codelab instructions after moving classes from "topeka-base" to "topeka-ui", as well as using the latest code of "complete" branch. It's also reported by a developer as [a comment](https://www.youtube.com/watch?v=9Jg1D07NgeI&lc=z13gd5fbqpvss5hpf04cirabgvm0sjxzxus0k) under thje I/O video.

The culprit is that topeka-base and topeka-ui are using different versions of the support library, therefore causing the incompatibility error.

The fix is simply use the same version of support library in topeka-ui’s gradle file. Something like:

```
...

ext {
supportLibVersion = “25.0.0”
}

dependencies {
compile “com.android.support:appcompat-v7:${supportLibVersion}”
compile “com.android.support:cardview-v7:${supportLibVersion}”
compile “com.android.support:design:${supportLibVersion}”
compile “com.android.support:recyclerview-v7:${supportLibVersion}”

api project(‘:topeka-base’)
}﻿
```

**Comment by desaibhavesh92** [comment]
Try this,

Change `android.support.v4.app.Fragment` to `android.app.Fragment`.

In my case its working.

**Comment by jkingyens** [comment]
thanks for reporting this. Should be fixed with d4dfdc8fe1f405b842b082f0e59580a58206f8f1
